up42/http/oauth.py

```diff
@@ -0,0 +1,59 @@
+import datetime as dt
+from dataclasses import dataclass
+
+import requests
+from requests import auth
+
+from up42.http import config, http_adapter
+
+
+@dataclass(eq=True, frozen=True)
+class Token:
+    access_token: str
+    expires_on: dt.datetime
+
+    @property
+    def has_expired(self):
+        return self.expires_on <= dt.datetime.now()
+
+
+class ProjectAuth(requests.auth.AuthBase):
+    def __init__(
+        self,
+        supply_credentials_settings=config.ProjectCredentialsSettings,
+        supply_token_settings=config.TokenProviderSettings,
+        create_adapter=http_adapter.create,
+    ):
+        credentials_settings = supply_credentials_settings()
+        token_settings = supply_token_settings()
+        self.client_id = credentials_settings.client_id
+        self.client_secret = credentials_settings.client_secret
+        self.token_url = token_settings.token_url
+        self.duration = token_settings.duration
+        self.timeout = token_settings.timeout
+        self.adapter = create_adapter(include_post=True)
```

Comment:
No, by default POST is not included. We need it for authentication, since backend-core is infamous for problematic token generation and it uses POST for it